src/data/trainer-config.ts

```diff
@@ -640,6 +672,36 @@ export class TrainerConfig {
     return this;
   }
 
+  /**
+   * Sets random pokemon from the trainers team to instant tera. Uses their specialty types is they have one.
+   * @param count The amount of pokemon to have instant tera
+   * @returns this
+   */
+  setRandomTeraModifiers(count: () => number): TrainerConfig {
+    this.genAIFuncs.push((party: EnemyPokemon[]) => {
+      const partyMemberIndexes = new Array(party.length).fill(null).map((_, i) => i);
+      for (let t = 0; t < Math.min(count(), party.length); t++) {
+        const randomIndex = randSeedItem(partyMemberIndexes);
+        partyMemberIndexes.splice(partyMemberIndexes.indexOf(randomIndex), 1);
+        if (this.specialtyTypes?.length) {
+          party[randomIndex].teraType = randSeedItem(this.specialtyTypes);
+        }
+        this.trainerAI.setInstantTera(randomIndex);
+      }
+    });
+    return this;
+  }
+
+  /**
+   * Sets a specific pokemon to instant tera
+   * @param index The index within the team to have instant tera
+   * @returns this
```

Comment:
NIT: Match existing docs:
```suggestion
   * @returns this {@linkcode TrainerConfig}
```